zebra/rt_netlink.c

```diff
@@ -300,9 +342,10 @@ static int netlink_route_change_read_unicast(struct sockaddr_nl *snl,
 	}
 
 	/* Route which inserted by Zebra. */
-	if (is_selfroute(rtm->rtm_protocol))
+	if (is_selfroute(rtm->rtm_protocol)) {
```

Comment:
While reviewing this I found out that this `is_selfroute()` function is not checking for `RTPROT_RIP` in addition to the other types.